### PR TITLE
Fix #239

### LINKED ISSUE #239: inplace operations on all cps causes update
I think it has something to do with python `__setitem__` syntax, but it'd be nice if we can avoid it

### PR DESCRIPTION
**Pull request by j042**
# Overview
This PR tries to remove some unnecessary copies

## Addressed issues
*  fixes #239

## Checklists
* [x] Documentations are up-to-date.
* [ ] Added example(s)
* [x] Added test(s)

### CODE REVIEW

File: splinepy/spline.py [hunk 1]
```diff
@@ -266,9 +266,64 @@ def _call_required_properties(spl, exclude="?"):
...
+    array: np.ndarray
+    key: str
+    coupled_key: str
+
+    Returns
+    -------
+    copied: bool
+      Whether copy was made
+    """
+    # is current value same is given array?
```
Comment (j042) [inline splinepy/spline.py:308]:
```suggestion
    # is current value same as given array?
```

**Review by j042 — COMMENTED** [review]
/wheels_all

File: .github/workflows/test_wheels.yml [hunk 2]
```diff
@@ -1,12 +1,12 @@
 name: test wheel builds
 
 on:
-  pull_request_review_comment:
-    types: [created, edited]
+  pull_request:
+    types: [labeled]
```
Comment (clemens-fricke) [inline .github/workflows/test_wheels.yml:5]: Did not know this is possible nice find.

File: splinepy/spline.py [hunk 3]
```diff
@@ -270,9 +270,76 @@ def _call_required_properties(spl, exclude="?"):
...
+
+def _safe_array_copy(spl, array, key, coupled_key):
+    """
+    In splinepy we make safe copies. For control_points and weights,
+    to support non-contiguous / homogeneous control points in backend,
+    we have a helper class called `PhysicalSpaceArray`. This syncs
+    numpy array and backend's control points through `ControlPointPointers`.
+
+    This function checks a non-copy criteria: self assignment.
+    This tends to happen during inplace operationrs.
```
Comment (clemens-fricke) [inline splinepy/spline.py:295]:
```suggestion
    This tends to happen during inplace operations.
```

**Review by clemens-fricke — APPROVED** [review]
From what I can see, this seems to be fine. 

Did not fully understand everything, but as I said it seems to be fine.

File: .github/workflows/test_wheels.yml (hunk 2 above)
Comment (j042) [reply .github/workflows/test_wheels.yml:5]: :D

**Review by clemens-fricke — APPROVED** [review]

**Review by jzwar — APPROVED** [review]
as discussed : LGTM, thanks

**Pull request merged**